recipes/swig/all/conanfile.py

```diff
@@ -0,0 +1,125 @@
+                    yield
+        else:
+            yield
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        deps_libpaths = self._autotools.library_paths
+        deps_libs = self._autotools.libs
+        deps_defines = self._autotools.defines
+        if self.settings.os == "Windows" and self.settings.compiler != "Visual Studio":
+            self._autotools.link_flags.append("-static")
+
+        libargs = list("-L\"{}\"".format(p) for p in deps_libpaths) + list("-l\"{}\"".format(l) for l in deps_libs)
+        args = [
+            "PCRE_LIBS={}".format(" ".join(libargs)),
+            "PCRE_CPPFLAGS={}".format(" ".join("-D{}".format(define) for define in deps_defines)),
+            "--host={}".format(self.settings.arch),
+            "--with-swiglibdir={}".format(self._swiglibdir),
+        ]
+        if self.settings.compiler == "Visual Studio":
+            self.output.warn("Visual Studio compiler cannot create ccache-swig. Disabling ccache-swig.")
+            args.append("--disable-ccache")
+            self._autotools.flags.append("-FS")
+
+        self._autotools.libs = []
+        self._autotools.library_paths = []
+
+        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def build(self):
+        self._patch_sources()
+        with tools.chdir(os.path.join(self._source_subfolder)):
+            self.run("./autogen.sh", win_bash=tools.os_info.is_windows)
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    def package(self):
+        self.copy(pattern="LICENSE*", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="COPYRIGHT", dst="licenses", src=self._source_subfolder)
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.install()
+
+        if self.settings.compiler != "Visual Studio":
+            with tools.chdir(os.path.join(self.package_folder, "bin")):
+                strip = tools.get_env("STRIP") or tools.which("strip")
+                ext = ".exe" if tools.os_info.is_windows else ""
+                if strip:
+                    self.run("{} swig{}".format(strip, ext), win_bash=tools.os_info.is_windows)
+                    self.run("{} ccache-swig{}".format(strip, ext), win_bash=tools.os_info.is_windows)
+
+    @property
+    def _swiglibdir(self):
+        return os.path.join(self.package_folder, "bin", "swiglib").replace("\\", "/")
+
+    def package_info(self):
+        # FIXME: Don't set cmake_find_package name because conan cmake generators do not define SWIG_EXECUTABLE
```

Comment:
Is this `<PKG-NAME>_EXECUTABLE` something standard in CMake? What happens if the sources provide more than one executable? Asking this just to know if it makes sense to add this as a feature in the CMake generators

Reply:
I's quite common:
- `SWIG_EXECUTABLE`: https://cmake.org/cmake/help/latest/module/FindSWIG.html
- `PYTHON_EXECUTABLE`: https://cmake.org/cmake/help/latest/module/FindPythonInterp.html
- `DOXYGEN_EXECUTABLE`: https://cmake.org/cmake/help/latest/module/FindDoxygen.html
- `PERL_EXECUTABLE`: https://cmake.org/cmake/help/latest/module/FindPerl.html
- `WGET_EXECUTABLE`: https://cmake.org/cmake/help/latest/module/FindWget.html

Also, some provide more than one _EXECUTABLE variable:
- `Java_JAVA_EXECUTABLE` `Java_JAVAC_EXECUTABLE` `Java_JAVAH_EXECUTABLE` ... https://cmake.org/cmake/help/latest/module/FindJava.html

But there are exceptions:
- `HTML_HELP_COMPILER`: https://cmake.org/cmake/help/latest/module/FindHTMLHelp.html

Also, it would be interesting to provide these as targets:
- `Doxygen::Doxygen` `Doxygen::dot` `Doxygen::mscgen` `Doxygen::dia`: https://cmake.org/cmake/help/latest/module/FindDoxygen.html
It is stated there that the `_EXECUTABLE` variables are getting deprecated.
(but conan should provide these for the sake of compatibility)